src/component/routes/public-route.component.jsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Route, Redirect } from "react-router-dom";
+import { useSelector } from "react-redux";
+
+const PublicRoute = ({ component: Component, ...rest }) => {
+  const isLogin = useSelector((state) => state.loginUser);
```

Comment:
naming conventions are also quite important. What does isLogin describe? I maybe misunderstood but a better and more understandable name for this would be isAuthenticated or isLoggedIn?